src/authentication/authentication.controller.ts

```diff
@@ -0,0 +1,98 @@
+import * as bcrypt from "bcrypt"
+import * as express from "express"
+import * as jwt from "jsonwebtoken"
+import User from "../models/User.interface"
+import UserWithThatEmailAlreadyExistsException from "../exceptions/UserWithThatEmailAlreadyExistsException"
+import WrongCredentialsException from "../exceptions/WrongCredentialsException"
+import Controller from "../interfaces/controller.interface"
+import DataStoredInToken from "../interfaces/dataStoredInToken"
+import TokenData from "../interfaces/tokenData.interface"
+import validationMiddleware from "../middleware/validation.middleware"
+import CreateUserDto from "../models/User.dto"
+import userModel from "./../models/User"
+import LogInDto from "./logIn.dto"
+import PasswordPolicyException from "../exceptions/PasswordPolicyException"
+
+class AuthenticationController implements Controller {
+    public path = "/auth"
+    public router = express.Router()
+    private user = userModel
+
+    constructor() {
+        this.initializeRoutes()
+    }
+
+    private initializeRoutes(): void {
+        this.router.post(`${this.path}/register`, validationMiddleware(CreateUserDto), this.registration)
+        this.router.post(`${this.path}/login`, validationMiddleware(LogInDto), this.loggingIn)
+        this.router.post(`${this.path}/logout`, this.loggingOut)
+    }
+
+    private registration = async (
+        request: express.Request,
+        response: express.Response,
+        next: express.NextFunction,
+    ): Promise<void> => {
+        const userData: CreateUserDto = request.body
+        if (await this.user.findOne({ email: userData.email })) {
+            next(new UserWithThatEmailAlreadyExistsException(userData.email))
```

Comment:
```suggestion
            throw new UserWithThatEmailAlreadyExistsException(userData.email)
```
analogicznie w pozostałych miejscach

Reply:
zrobione